sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthRSA256Service.java

```diff
@@ -68,6 +80,82 @@ public AuthRSA256Service(RsaPublicKeyLoader rsaPublicKeyLoader, long leeway) {
     }
   }
 
+  /**
+   * Validates, that the jwt token issuer is the same as the attribute {@link
+   * LoadedPublicKey#getRequiredIssuer()}. Further it will be validated, that the issuer Url has the
+   * same base as the configured {@link KeySource} source uri, that depends on the implementation.
+   *
+   * @param decodedJWT the decoded json web token.
+   * @param loadedPublicKey the structure with all correlated attributes regarding the key.
+   * @return <code>true</code> if the jwt issuer is the same as the configured requiredIssuer and
+   *     the baseUrl is the same as the {@link KeySource} uri or the configuration attribute
+   *     requiredIssuer is not set. Otherwise <code>false</code> is returned.
+   */
+  private boolean validateIssuer(DecodedJWT decodedJWT, LoadedPublicKey loadedPublicKey) {
+
+    if (StringUtils.isNotBlank(loadedPublicKey.getRequiredIssuer())
+        && (!StringUtils.equals(decodedJWT.getIssuer(), loadedPublicKey.getRequiredIssuer())
+            || !validateIssuerAgainstKeySourceUrl(
+                decodedJWT.getIssuer(), loadedPublicKey.getKeySource()))) {
```

Comment:
* I'd prefer an early exit in case the required issuer is blank to simplify this statement.
* What's the benefit of `StringUtils#equals` here? Can't we just use `loadedPublicKey.getRequiredIssuer().equals(...)`?